website/public/js/services/achievementServices.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+/**
+ * Services that handle achievement logic.
+ */
+
+angular.module("habitrpg").factory("Achievement",
+['$rootScope', function($rootScope) {
+
+	function openModal(achievementName) {
+		$rootScope.openModal('achievements/' + achievementName,
+			{controller:'UserCtrl', size:'sm'});
+	}
+
+	return {
+		openModal: openModal
```

Comment:
Let's call this function something like `displayAchievement`. That way, if we change the inner works of how we're displaying the achievements, the function name isn't tied to the modal process.